Makefile

```diff
@@ -95,6 +95,11 @@ tests:
 
 vet:
 	go vet ./...
+	go tool vet --shadow ./test
```

Comment:
I insist that leaving this out is asking for more work. Feel free to commit the other changes here if you want, but I won't remove this.